src/com/parse4cn1/command/ParseUploadCommand.java

```diff
@@ -54,6 +54,7 @@ void setUpRequest(ConnectionRequest request) throws ParseException {
 
         if (contentType != null) {
             request.addRequestHeader(ParseConstants.HEADER_CONTENT_TYPE, contentType);
+            addHeader(ParseConstants.HEADER_CONTENT_TYPE, contentType);
```

Comment:
On a second thought, a more stable solution would only to overwrite the request headers that have not been explicitly set. In that way, it would work not just for ParseUploadCommand but any other (future) commands where the CONTENT_TYPE is explicitly set... What do you think?